R/FilterStates.R

```diff
@@ -622,7 +622,11 @@ FilterStates <- R6::R6Class( # nolint
           # and this no longer needs to be a function to pass sid. reactive in the FilterState
           # is also beneficial as it can be cached and retriger filter counts only if
           # returned vector is different.
-          x_reactive = reactive(data_reactive(sid)[, x$varname, drop = TRUE]),
+          x_reactive = if (attr(state, "count_type") == "none") {
+            reactive(NULL)
+          } else {
+            reactive(data_reactive(sid)[, x$varname, drop = TRUE])
+          },
```

Comment:
```suggestion
          x_reactive = 
          if (attr(state, "count_type") == "none") {
            reactive(NULL)
          } else if (attr(state, "count_type") == "all") {
            reactive(data_reactive(sid)[, x$varname, drop = TRUE])
          } else {
            stop("hierarchical filters are not supported yet")
          },
```

Reply:
Adding handling of remaining cases, after consulting with @gogonzo.
We can now regulate the default behavior by changing the default value of `count_type` in `filter_settings`.